From aaaf215aa99f65e72d413646efb15a950f694a00 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Sun, 29 Jan 2017 20:59:15 +0200
Subject: [PATCH 0913/1345] dts: a70x0: add device tree configuration for
 board setup E

commit  669550a9be03f11a26023addc776abf4d50b791b from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

A7040 DB board is a modular board,
which supports multiple possible configurations.
This patch introduce predefined board topology for setup E.

- Add device tree for E setup
	- AP_UART, AP_SD, AP_CP0
	- CP0_RGMII0, CP0_SDIO, CP0_I2C0, CP0_I2C1
	  CP0_SATA1, CP0-SATA0, CP0_SGMII0

Change-Id: Ic4c24869e480a0bf4d080e4190ad03efef4c6d66
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37649
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 .../mvebu/a7k-a8k/armada7040-db-setup.txt          |  123 ++++++++------
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 arch/arm64/boot/dts/marvell/armada-7040-db-E.dts   |  179 ++++++++++++++++++++
 4 files changed, 251 insertions(+), 55 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-7040-db-E.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 805cc00..388c385 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -29,6 +29,9 @@ Boards:
    - "marvell,armada7040-db-C", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
+   - "marvell,armada7040-db-E", "marvell,armada7040-db"
+      when the board being used is the Armada 7040 Development board
+
    - "marvell,armada7040-db-nand", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
diff --git a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
index bba3b5f..fb35e37 100644
--- a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
@@ -20,18 +20,18 @@ Board Setup
 ------------
 For A7040 A revision SoC - Please use armada-7040-db-A configuration,
 
-|		| 	1: A		|	2:B		|	3:C		|
-|---------------|-----------------------|-----------------------|-----------------------|
-|Device tree	|armada-7040-db-A.dtb	|armada-7040-db-B.dtb	|armada-7040-db-C.dtb	|
-|CP0-SW1:1	| ON			| OFF			| ON			|
-|CP0-SW1:2	| OFF			| ON			| OFF			|
-|CP0-SW1:3	| ON			| ON			| ON			|
-|CP0-SW1:4	| OFF			| ON			| OFF			|
-|CP0-SW1:5	| OFF			| ON			| OFF			|
-|CP0-SW1:6	| ON			| ON			| ON			|
-|CP0-SW1:7	| ON			| ON			| ON			|
-|CP0-SW1:8	| OFF			| OFF			| OFF			|
------------------------------------------------------------------------------------------
+|		| 	1: A		|	2:B		|	3:C		|       4:E             |
+|---------------|-----------------------|-----------------------|-----------------------|-----------------------|
+|Device tree	|armada-7040-db-A.dtb	|armada-7040-db-B.dtb	|armada-7040-db-C.dtb	|armada-7040-db-E.dtb	|
+|CP0-SW1:1	| ON			| OFF			| ON			| OFF 			|
+|CP0-SW1:2	| OFF			| ON			| OFF			| OFF 			|
+|CP0-SW1:3	| ON			| ON			| ON			| ON			|
+|CP0-SW1:4	| OFF			| ON			| OFF			| OFF			|
+|CP0-SW1:5	| OFF			| ON			| OFF			| OFF			|
+|CP0-SW1:6	| ON			| ON			| ON			| ON			|
+|CP0-SW1:7	| ON			| ON			| ON			| ON			|
+|CP0-SW1:8	| OFF			| OFF			| OFF			| OFF			|
+-----------------------------------------------------------------------------------------------------------------
 
 DPR Setting
 ------------
@@ -44,30 +44,35 @@ DPR Setting
 | 					| DPR2,3,4,5 - 1-2 position		 | DPR2,3,4,5 - 2-3 position              |
 ----------------------------------------|----------------------------------------|----------------------------------------|
 
-|					| 	3: C				 |
-|---------------------------------------|----------------------------------------|
-| Mpp Jumper Module (SLM1548)		| DPR133 - 143  2-3 position 		 |
-| Serdess Jumper Module (SLM1547)	| DPR152,153,154,155 - 1-2 position	 |
-|					| DPR148,149,150,151 - 1-2 position	 |
-| CP0 Serdess module (SLM1521)		| SW1 - (1,3,6,7)= ON,  (2,4,5,8)=OFF	 |
-| 					| DPR2,3,4,5 - 1-2 position		 |
-| Install TDM Module (SLM1448)		| on DB-7040 CON46			 |
-----------------------------------------|----------------------------------------|
+|					| 	3: C				 |		4:E                                    |
+|---------------------------------------|----------------------------------------|-----------------------------------------------------|
+| Mpp Jumper Module (SLM1548)		| DPR133 - 143  2-3 position		 | DPR133 -143  1-2 position , DPR144-147 2-3 position |
+| Serdess Jumper Module (SLM1547)	| DPR152,153,154,155 - 1-2 position	 | DPR152,153,154,155 - 1-2 position                   |
+|					| DPR148,149,150,151 - 1-2 position	 | DPR148,149,150,151 - 1-2 position                   |
+| CP0 Serdess module (SLM1521)		| SW1 - (1,3,6,7)= ON,  (2,4,5,8)=OFF	 | SW1 - (3,6,7)= ON,  (1,2,4,5,8)=OFF                 |
+| 					| DPR2,3,4,5 - 1-2 position		 | DPR2,3,4,5 - 1-2 position                           |
+| Install TDM Module (SLM1448)		| on DB-7040 CON46			 | -						       |
+|					|					 |                                                     |
+| Install Audio Module (SLM1449)	| - 					 | -                                                   |
+|					|					 |                                                     |
+| DB-4070 Modular (SLM1546)		| - 					 | DPR53-54 1-2 position , DPR61 2-3 position          |
+|					| 					 | DPR10-11 position 1-2, DPR12-15 position 1-2        |
+----------------------------------------|----------------------------------------|-----------------------------------------------------|
 
 The tables below summarizes the interface configuration of each setup
 
 SerDes Configuration
 ------------------------
 
-| CP0 Lane 	| 	1: A		|	2: B		|       3: C            |
-|---------------|-----------------------|-----------------------|-----------------------|
-| 0		| SGMII2		|	PICE0 (x4)	|       PCIE0 (x1)      |
-| 1		| USB3_HOST0		|	PCIE0 (x4)	|       USB3_HOST0      |
-| 2		| SFI (10G)		|	PCIE0 (x4)	|       SFI (10G)      	|
-| 3		| SATA1			|	PCIE0 (x4)	|       SATA1		|
-| 4		| USB3_HOST1		|	USB3_HOST1	|       USB3_HOST1      |
-| 5		| PCIE2 (x1)		|	PCIE2 (x1)	|       PCIE2 (x1)      |
------------------------------------------------------------------------------------------
+| CP0 Lane 	| 	1: A		|   2: B	| 3: C            | 4:E		  |
+|---------------|-----------------------|---------------|-----------------|---------------|
+| 0		| SGMII2		| PICE0 (x4)	| PCIE0 (x1)      |  PCIE0 (x1)   |
+| 1		| USB3_HOST0		| PCIE0 (x4)	| USB3_HOST0      |  SATA0        |
+| 2		| SFI (10G)		| PCIE0 (x4)	| SFI (10G)       |  SGMII0 (1G)  |
+| 3		| SATA1			| PCIE0 (x4)	| SATA1		  |  SATA1        |
+| 4		| USB3_HOST1		| USB3_HOST1	| USB3_HOST1      |  USB3_HOST1   |
+| 5		| PCIE2 (x1)		| PCIE2 (x1)	| PCIE2 (x1)      |  PCIE2 (x1)   |
+-------------------------------------------------------------------------------------------
 
 - USB2_H0 (UTMI only)
 - USB2_H1 (UTMI & USB3 SerDes)
@@ -76,30 +81,31 @@ SerDes Configuration
 Multi-purpose pin configurations
 --------------------------------
 
-Setup  #	|	1	|	2	|       3       |
-Name   #	|  	A 	|	B	|       C       |
------------------------------------------------------------------
-AP806:				|		|		|
------------------------------------------------------------------
-	AP-SDIO | [0-5]		| [0-5]		| [0-5]         |
-	AP-UART0| [11,19]	| [11,19]	| [11,19]       |
------------------------------------------------------------------
-CP:				|		|		|
------------------------------------------------------------------
-	CP-SPI0	| N/C		| N/C		| [6-11]	|
-	CP-SPI1	| [13-16]	| [13-16]	| [13-16]	|
-	NAND	| N/C		| N/C		| N/C		|
-	RGMII0	| N/C		| [44-55]	| [0-11]	|
-	RGMII1	| [0-11]	| [0-11]	| [44-55]	|
-	CP-UART0| [29-30]	| [29-30]	| [29-30	|
-	CP-UART1| N/C		| N/C		| N/C		|
-	SD	| [56-61]	| [56-61]	| [56-61]	|
-	GPIO	| [62]		| [62]		| [62]		|
-	TDM	| N/C		| N/C		| [0-5]		|
-	TWSI	| [37-38]	| [37-38]	| [37-38]	|
-	SATA0	| N/C		| N/C		| N/C		|
-	SATA1	| [28]		| [28]		| [28]		|
------------------------------------------------------------------
+Setup  #	|	1	|	2	|       3       |	4	|
+Name   #	|  	A 	|	B	|       C       |	E	|
+---------------------------------------------------------------------------------
+AP806:				|		|		|		|
+---------------------------------------------------------------------------------
+	AP-SDIO | [0-5]		| [0-5]		| [0-5]         | [0-3]		|
+	AP-SPI0 | N/C		| N/C		| N/C		| [4,5]		|
+	AP-UART0| [11,19]	| [11,19]	| [11,19]       | [11,19]	|
+---------------------------------------------------------------------------------
+CP:				|		|		|		|
+---------------------------------------------------------------------------------
+	CP-SPI0	| N/C		| N/C		| [6-11]	| N/C		|
+	CP-SPI1	| [13-16]	| [13-16]	| [13-16]	| [13-16]	|
+	NAND	| N/C		| N/C		| N/C		| N/C		|
+	RGMII0	| N/C		| [44-55]	| [0-11]	| [0-11]	|
+	RGMII1	| [0-11]	| [0-11]	| [44-55]	| N/C		|
+	CP-UART0| [29-30]	| [29-30]	| [29-30	| [29-30|	|
+	CP-UART1| N/C		| N/C		| N/C		| [47,49-51]	|
+	SD	| [56-61]	| [56-61]	| [56-61]	| [54,56-61]	|
+	GPIO	| [62]		| [62]		| [62]		| [62]		|
+	TDM	| N/C		| N/C		| [0-5]		| N/C		|
+	TWSI	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
+	SATA0	| N/C		| N/C		| N/C		| N/C		|
+	SATA1	| [28]		| [28]		| [28]		| [28]		|
+---------------------------------------------------------------------------------
 
 Network configuration
 ---------------------
@@ -118,9 +124,16 @@ Network configuration
  eth1           | CP0   | eth1          | 3             | RGMII1                        |
  ----------------------------------------------------------------------------------------
 
- 3: C:
+3: C:
  Interface      | CP#   | PPv2 Port     | GMAC Port     | Board Interface               |
  ---------------|-------|---------------|---------------|-------------------------------|
  eth0           | CP0   | eth0          | 0             | SFI (10G)                     |
  eth1           | CP0   | eth1          | 3             | RGMII1                        |
  ----------------------------------------------------------------------------------------
+
+4: E:
+ Interface      | CP#   | PPv2 Port     | GMAC Port     | Board Interface               |
+ ---------------|-------|---------------|---------------|-------------------------------|
+ eth0           | CP0   | eth0          | 0             | SGMII0 (1G)                   |
+ eth1           | CP0   | eth1          | 2             | RGMII0                        |
+ ----------------------------------------------------------------------------------------
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 38f9ec1..fae7a7e 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -13,6 +13,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-C.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-E.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-nand.dtb
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
new file mode 100644
index 0000000..9a19b61
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7040 development board
+ * This board file supports the E configuration of the board
+ */
+
+#include "armada-7040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-7040 development board E setup";
+	compatible = "marvell,armada7040-db-E", "marvell,armada7040-db", "marvell,armada7040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	ap806 {
+		config-space {
+			sdhci@6e0000 {
+				status = "okay";
+			};
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "okay";
+				clock-frequency = <100000>;
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			serial@702000 {
+				status = "disabled";
+			};
+			serial@702100 {
+				status = "disabled";
+			};
+			serial@702200 {
+				status = "disabled";
+			};
+			serial@702300 {
+				status = "disabled";
+			};
+			sata: sata@540000 {
+				status = "okay";
+
+				sata-port@0 {
+					phys = <&comphy0 1 COMPHY_SATA0>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+				sata-port@1 {
+					phys = <&comphy0 3 COMPHY_SATA1>;
+					phy-names = "comphy";
+					status = "okay";
+				};
+			};
+			usb3h0: usb3@500000 {
+				status = "disabled";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			u3d@520000 {
+				status = "disabled";
+			};
+			udc@524100 {
+				status = "disabled";
+			};
+			sdhci@780000 {
+				status = "okay";
+			};
+			spi@700680 {
+				status = "okay";
+			};
+			i2c1: i2c@701000 {
+				status = "okay";
+				clock-frequency = <100000>;
+			};
+			i2c2: i2c@701100 {
+				status = "okay";
+			};
+
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "sgmii";
+					phys = <&comphy0 2 COMPHY_SGMII0>;
+					phy-names = "comphy";
+				};
+				emac2: mac2 {
+					phy-mode = "rgmii";
+					phy = <&phy2>;
+				};
+				emac3: mac3 {
+					phy-mode = "none";
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "okay";
+				};
+				eth2: eth2@030000 {
+					status = "disabled";
+				};
+			};
+			eip197: eip197@800000 {
+				status = "okay";
+			};
+			comphy {
+				status = "okay";
+			};
+		};
+
+		pcie@0x640000 {
+			status = "okay";
+		};
+		pcie@0x620000 {
+			status = "disabled";
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+	};
+};
-- 
1.7.9.5

